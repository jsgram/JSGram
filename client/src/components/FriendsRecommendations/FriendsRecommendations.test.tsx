import React from 'react';
import {FriendsRecomendations} from './index';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import * as reactstrap from 'reactstrap';

describe('FriendsRecommendations component', () => {
    let renderer: any;

    const props = {
        loggedUsername: 'somevalue',
        friendsRecommendations: {
            users: [{
                photoPath: 'somevalue',
                _id: 'somevalue',
                username: 'somevalue',
            }],
            loading: false,
        },
        followUser: jest.fn(() => ('somevalue')),
    };

    const initialState = {
        profileEdit: {
            newUsername: 'somevalue',
        },
        feed: {
            loggedUsername: 'somevalue',
        },
        search: {
            searchResults: [],
        },
    };
    const mockStore = configureStore();

    beforeEach(() => {
        reactstrap.Button = jest.fn(() => <div></div>);
        const store = mockStore(initialState);
        renderer = mount(<Provider store={store}>
            <BrowserRouter><FriendsRecomendations {...props}/></BrowserRouter>
        </Provider>);
    });

    test('render-success', () => {
        expect(renderer.html()).toHaveLength(349);
    });
});
