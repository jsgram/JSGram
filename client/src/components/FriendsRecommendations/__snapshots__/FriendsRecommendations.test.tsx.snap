// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendsRecommendations component render-success 1`] = `ReactWrapper {}`;
